### PR TITLE
Material: Problems editing Quantity properties

### LINKED ISSUE #13020: Material editor input fields show zero and non-editable values for non-zero properties
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

When I open the material editor, select some material and double-click on one of its defined properties, the input field shows zeros even though the property is non-zero. If I enter a value, nothing changes.

zero properties

@davesrocketshop FYI

### Full version info

```shell
OS: Windows 10 build 19045
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.36391 (Git)
Build type: Release
Branch: main
Python 3.10.13, Qt 5.15.8, Coin 4.0.2, Vtk 9.2.6, OCC 7.6.3
Locale: Polish/Poland (pl_PL)
```

### Subproject(s) affected?

Other (specify in description)

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by davesrocketshop**
There were several issues here, not just one. The following are fixed:

    Incorrect display of Quantity items (NaN)
    Editing and updating quantity items
    Editing and updating items on the first row.

There are still issues with editing lists, but these were known issues at the time of initial merge. This has been split out into issue #13435

fixes #13020

### CODE REVIEW

**Comment by FEA-eng** [comment]
@davesrocketshop Can you update the branch? The tests should work now.

File: src/Mod/Material/Gui/BaseDelegate.cpp [hunk 1]
```diff
@@ -354,10 +354,11 @@ void BaseDelegate::setModelData(QWidget* editor,
         value = chooser->fileName();
     }
     else if (type == Materials::MaterialValue::Quantity) {
-        auto input = dynamic_cast<Gui::InputField*>(editor);
+        auto input = dynamic_cast<Gui::QuantitySpinBox*>(editor);
```
Comment (chennes) [inline src/Mod/Material/Gui/BaseDelegate.cpp:357]: It's not important to fix here, but some static analyzers will note the use of dynamic_cast without checking for a null result. You could combine the variable definition into the if clause to do that check without actually adding any lines of code.

Comment (davesrocketshop) [reply src/Mod/Material/Gui/BaseDelegate.cpp:357]: Good point. I'll be back in there for other things so I'll fix it then.

**Review by chennes — APPROVED** [review]
I gave this a test run and it does fix the problems identified in the linked Issue.

**Pull request merged**